src/basic/env-util.c

```diff
@@ -395,6 +394,24 @@ int strv_env_replace(char ***l, char *p) {
         return 1;
 }
 
+int strv_env_replace_strdup(char ***l, const char *assignment) {
+        char *p;
+        int r;
+
+        /* Like strv_env_replace(), but copies the argument. */
+
+        p = strdup(assignment);
+        if (!p)
+                return -ENOMEM;
+
+        r = strv_env_replace(l, p);
```

Comment:
It may retrun `-EINVAL` when the input string does not contain `=`, and in that case, `p` is not consumed...

Reply:
Duh, I even put a TAKE_PTR there, but forgot about `_cleanup_`. Fixed now.
(And also the unused var flagged by clang.)